enumerables.rb

```diff
@@ -0,0 +1,75 @@
+module Enumerable
+  def my_each
+    length.times { |value| yield(self[value]) }
+    self
+  end
+
+  def my_each_with_index
+    length.times { |value| yield(self[value], value) }
+    self
+  end
+
+  def my_select
+    array = []
+    my_each { |num| array << num if yield(num) }
+    array
+  end
```

Comment:
- [ ] `my_select` is not working as the original `select` method, also is not passing all test cases.

### `select` is an Enumerable that has these characteristics:
- Returns an Enumerator if no block is given.
```
[1, 2, 3].select
# #<Enumerator: [1, 2, 3]:select>
```
- Returns a **new array** with the results that match the condition in the block
```
[1, 2, 3].select { |num| num > 2 }
# [3]
```
- doesn't change/modify the original array/range/hash

[Enumerable#select](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-select)